hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/impl/AWSRegionEndpointInformation.java

```diff
@@ -0,0 +1,70 @@
+package org.apache.hadoop.fs.s3a.impl;
+
```

Comment:
@steveloughran starting to work on the CRT changes, wanted to get your opinion on this. What do you think about doing some refactoring around this region logic, so we have our own builder and container class for `AWSRegionEndpointInformation` which all clients can get information out of. 

Without this you have to duplicate the endpoint region logic for CRT and the other clients since they use different builder classes. CRT uses `S3CrtAsyncClientBuilder` which does not extend `S3BaseClientBuilder` and so you can't use the existing method..

Reply:
Pulling out is good.

we need to fix that region resolution logic

* work in ec2/k8s without going out of region -automatically
* add an option to use the default chain without the "add a space" trick, e.g a region called "(sdk)". Not backwards compatible, but makes clear what happens.
*